mantidproject/mantid
# Move the ISIS SANS Interface Out of the Framework.

### Description of work

#### Summary of work
Moves the main content of the ISIS SANS GUI out of the framework and into the `mantidqtinterfaces` package. This removes a lot of modules that cannot be run being available in the `mantid` package where Qt is not available. 

Fixes #37952 

#### Further detail of work
The move required some changes to the packages, as there was code in the `gui_logic` that was relied upon by other elements in the `sans` reduction package. This should now be fixed so the imports only occur one-way. 

### To test:
1. Open the ISIS SANS Interface.
2. Load the user file and batch file from the `loqdemo` in the training course data. 
3. Process All
4. Ensure the reduction completes successfully. 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
